rfcs/active/0000-badge.md

```diff
@@ -0,0 +1,362 @@
+  --oui-badge-color-success-text: var(--oui-color-success-contrast, #ffffff);
+  --oui-badge-color-danger-bg: var(--oui-color-danger, #dc3545);
+  --oui-badge-color-danger-text: var(--oui-color-danger-contrast, #ffffff);
+  --oui-badge-color-warning-bg: var(--oui-color-warning, #ffc107);
+  --oui-badge-color-warning-text: var(--oui-color-warning-contrast, #000000);
+  --oui-badge-color-info-bg: var(--oui-color-info, #0dcaf0);
+  --oui-badge-color-info-text: var(--oui-color-info-contrast, #000000);
+
+  /* Status indicators */
+  --oui-badge-color-status-success: var(--oui-color-success, #52c41a);
+  --oui-badge-color-status-processing: var(--oui-color-info, #1890ff);
+  --oui-badge-color-status-default: var(--oui-color-secondary, #d9d9d9);
+  --oui-badge-color-status-error: var(--oui-color-danger, #ff4d4f);
+  --oui-badge-color-status-warning: var(--oui-color-warning, #faad14);
+
+  /* Ribbon specific */
+  --oui-badge-ribbon-height: 1.5rem;
+  --oui-badge-ribbon-padding-x: 0.5rem;
+  --oui-badge-ribbon-corner-size: 0.5rem;
+}
+```
+
+## Behavior
+
+### Sizing and Positioning
+
+Badges should automatically scale to match the size of the parent element when used inline (e.g., with headings or buttons). This will be achieved through using relative font sizes (`em` units).
+
+When the `position` attribute is used, the badge will be absolutely positioned relative to its parent container, which should have `position: relative` applied.
+
+### Accessibility Considerations
+
+1. **Color contrast**: All badge color variants must meet WCAG 2.1 AA contrast requirements.
+2. **Screen readers**:
+   - When a badge contains only a number (e.g., "4" for notifications), an `aria-label` should be provided to give context (e.g., "4 unread messages").
+   - When positioned badges are used as status indicators without text, `aria-label` must be provided.
+3. **Focus**: Badges themselves are not focusable elements unless they function as interactive elements.
+4. **Internationalization**:
+   - Badge text should be translatable
+   - The component should support RTL languages, which affects positioning variants
+
+## Use Cases
+
+1. **Notification indicators**: Showing counts of unread messages, notifications, etc.
+2. **Status indicators**: Displaying the state of an item (new, updated, deprecated, success, error)
+3. **Categorization**: Labeling content with categories or tags
+4. **Highlighting**: Drawing attention to new features or content
+5. **Numeric counters**: Displaying metrics, scores, or numerical values
+6. **Progress indicators**: Showing steps in a multi-step process (e.g., "1 of 3")
+7. **Dot indicators**: Small visual cues to indicate status without textual content
+8. **Ribbons**: Corner labels or banners to highlight special status or promotions
+9. **Standalone badges**: Used independently of other components to convey information
+10. **Interactive badges**: Clickable elements that trigger actions or reveal additional information
```

Comment:
I wonder if it would be helpful to prohibit badges themselves from being interactive, and recommending to nest inside of a `button` when interactivity is needed, letting the button do the interactivity.

That way, badge also doesn't need to be concerned with keyboard behaviour, accessible names, etc.